lib/src/util/bolt_helpers.cc

```diff
@@ -0,0 +1,195 @@
+    // false otherwise. err_msg contains the most recent http_file_download_exception's error
+    // message; it is initially empty.
+    std::tuple<bool, std::string> downloadFileWithCurl(const std::vector<std::string>& master_uris,
+                                                        uint32_t connect_timeout_s,
+                                                        uint32_t timeout_s,
+                                                        lth_curl::client& client,
+                                                        const fs::path& file_path,
+                                                        const lth_jc::JsonContainer& uri) {
+        auto endpoint = createUrlEndpoint(uri);
+        std::tuple<bool, std::string> result = std::make_tuple(false, "");
+        for (auto& master_uri : master_uris) {
+            auto url = master_uri + endpoint;
+            lth_curl::request req(url);
+
+            // Request timeouts expect milliseconds.
+            req.connection_timeout(connect_timeout_s*1000);
+            req.timeout(timeout_s*1000);
+
+            try {
+                lth_curl::response resp;
+                client.download_file(req, file_path.string(), resp, NIX_TASK_FILE_PERMS);
+                if (resp.status_code() >= 400) {
+                    throw lth_curl::http_file_download_exception(
+                        req,
+                        file_path.string(),
+                        lth_loc::format("{1} returned a response with HTTP status {2}. Response body: {3}", url, resp.status_code(), resp.body()));
+                }
+            } catch (lth_curl::http_file_download_exception& e) {
+                // Server-side error, do nothing here -- we want to try the next master-uri.
+                LOG_WARNING("Downloading the task file from the master-uri '{1}' failed. Reason: {2}", master_uri, e.what());
+                std::get<1>(result) = e.what();
+            } catch (lth_curl::http_request_exception& e) {
+                // For http_curl_setup and http_file_operation exceptions
+                throw Module::ProcessingError(lth_loc::format("Downloading the task file failed. Reason: {1}", e.what()));
+            }
+
+            if (fs::exists(file_path)) {
+                std::get<0>(result) = true;
+                return result;
+            }
+        }
+
+        return result;
+    }
+
+    std::string createUrlEndpoint(const lth_jc::JsonContainer& uri) {
+        std::string url = uri.get<std::string>("path");
+        auto params = uri.getWithDefault<lth_jc::JsonContainer>("params", lth_jc::JsonContainer());
+        if (params.empty()) {
+            return url;
+        }
+        auto curl_handle = lth_curl::curl_handle();
+        url += "?";
+        for (auto& key : params.keys()) {
+            auto escaped_key = std::string(
+            lth_curl::curl_escaped_string(curl_handle, key));
```

Comment:
Is there a reason these lines are broken at the `(` the way they are? They look short enough without being wrapped.

Reply:
No, I think you're correct that there's no need for the broken (